core/src/main/java/org/apache/accumulo/core/fate/accumulo/StatusMappingIterator.java

```
@@ -0,0 +1,148 @@
+ * iterator allows for checking of the status column's value against a set of acceptable statuses
+ * within a conditional mutation.
+ */
+public class StatusMappingIterator implements SortedKeyValueIterator<Key,Value> {
+
+  private static final String PRESENT = "present";
+  private static final String ABSENT = "absent";
+  private static final String STATUS_SET_KEY = "statusSet";
+
+  private SortedKeyValueIterator<Key,Value> source;
+  private final Set<String> acceptableStatuses = new HashSet<>();
+  private Value mappedValue;
+
+  /**
+   * The set of acceptable must be provided as an option to the iterator using the
+   * {@link #STATUS_SET_KEY} key.
+   */
+  @Override
+  public void init(SortedKeyValueIterator<Key,Value> source, Map<String,String> options,
+      IteratorEnvironment env) throws IOException {
+    this.source = source;
+    if (options.containsKey(STATUS_SET_KEY)) {
+      String[] statuses = decodeStatuses(options.get(STATUS_SET_KEY));
+      acceptableStatuses.addAll(Arrays.asList(statuses));
+    }
+  }
+
+  @Override
+  public boolean hasTop() {
+    return source.hasTop();
+  }
+
+  @Override
+  public void next() throws IOException {
+    source.next();
+    mapValue();
+  }
+
+  @Override
+  public void seek(Range range, Collection<ByteSequence> columnFamilies, boolean inclusive)
+      throws IOException {
+    source.seek(range, columnFamilies, inclusive);
+    mapValue();
+  }
+
+  /**
+   * Maps the value of the status column to "present" or "absent" based on its presence within the
+   * set of statuses.
+   */
+  private void mapValue() {
+    if (source.hasTop()) {
+      String currentValue = source.getTopValue().toString();
+      mappedValue =
+          acceptableStatuses.contains(currentValue) ? new Value(PRESENT) : new Value(ABSENT);
+    }
+  }
+
+  @Override
+  public Key getTopKey() {
+    return source.getTopKey();
+  }
+
+  @Override
+  public Value getTopValue() {
+    return mappedValue;
```

This makes things fail fast when mappedValue is not set

```suggestion
    return Objects.requireNonnull(mappedValue);
```